src/parse.cpp

```diff
@@ -849,6 +849,16 @@ expression_ptr parser::parse_object_literal() {
       QLJS_PARSER_UNIMPLEMENTED();
     }
     switch (this->peek().type) {
+    case token_type::complete_template:
+    case token_type::incomplete_template:
```

Comment:
Blocking: If we encounter an `incomplete_template`, we should parse the template literal using e.g. `this->parse_template`. Otherwise, parsing will get out of sync, and eventually we'll treat the end of the template literal as the start of a different template literal.

Your tests don't trigger the `incomplete_template` code path. If they did, I think you'd see the problem I'm talking about. Example:

```js
({ `hello${k}`: v })
// ^^^^^^^^ incomplete_template
```